app/models/user.rb

```diff
@@ -21,10 +22,16 @@ def set_default_alts
   class << self
 
     def find_or_create_by_oauth(params)
-      user = User.find_or_initialize_by(uid: params['uid'])
+      user = User.find_or_initialize_by(uid: params['uid'], provider: params['provider'])
+      census_roles = (params['info']['roles'] || []).map { |role| role['name'] }
+
+      user.role = 'student'
+      user.role = 'reviewer' if census_roles.include?('staff')
+      user.role = 'admin' if census_roles.include?('admin')
+
```

Comment:
This is awesome!! It'll make things much easier & possibly eliminate the need for admin to be able to edit users; one thing to consider is that currently, you cannot be a reviewer if you are an admin. Who is classified as an admin? Do those people ever want the opportunity to be a reviewer?

Reply:
Yes! The idea is that (eventually) most (all?) user roles will derive from roles in Census. TBD how we deal with situations like this one where specific roles (like reviewer) don't exist in Census. We can do this kind of mapping or we could go the direction where when we want a new role, we make it in Census (in this case, census would get a full-circle-reviewer role).

Re: admin / reviewer, we could solve that in Full Circle by making users have many roles instead of a single one. Let's punt for now!

Reply:
Asking for clarification in https://turingschool.slack.com/archives/G6VCACMBN/p1521152283000121.

Reply:
Per slack convo and https://trello.com/c/k8ANMiZF/244-update-census-to-have-full-circle-roles we're going with full circle roles in Census to make permissions more granular.